src/net/posix-stack.cc

```diff
@@ -730,15 +730,21 @@ class posix_udp_channel : public udp_channel_impl {
         char* _buffer;
         cmsg_with_pktinfo _cmsg;
 
-        recv_ctx() {
+        recv_ctx(bool use_pktinfo) {
```

Comment:
> why in a unix-domain datagram socket that has non-empty ancillary data you decide to zero msg_controllen and basically hide information that exists.

Hmm can a unix-domain datagram socket have a non-empty ancillary data if I never request them?
I decided to zero `msg_controllen` before the call to `recvmsg`, and thus I'm asking the kernel not to copy not needed ancillary data to the userspace.

From [man 2 recvmsg](https://www.man7.org/linux/man-pages/man2/recvmsg.2.html):
>       When recvmsg() is called,
>       msg_controllen should contain the length of the available buffer
>       in msg_control; upon return from a successful call it will
>       contain the length of the control message sequence.